hclsyntax/parser.go

```diff
@@ -867,8 +867,9 @@ Traversal:
 						Collection: ret,
 						Key:        keyExpr,
 
-						SrcRange:  rng,
-						OpenRange: open.Range,
+						SrcRange:     hcl.RangeBetween(from.Range(), rng),
```

Comment:
`from` in this function is the expression that the subsequent postfix operators are applying to, so in an expression like `foo[bar]` the `from` variable refers to the `foo` expression.